app/models/grouping.rb

```diff
@@ -749,17 +749,16 @@ def marking_state(result)
   end
 
   def get_total_test_script_marks
-    total = 0
 
     #find the unique test scripts for this submission
-    test_script_ids = test_script_results.pluck(:id).uniq
+    test_script_ids = test_script_results.pluck(:test_script_id).uniq
 
     #add the latest result from each of our test scripts
     test_script_ids.sum do |test_script_id|
-      last_mark = self.test_script_results
-                      .where(test_script_id: test_script_id)
-                      .last
-      last_mark.nil? ? 0 : last_mark.marks_earned
+      last_result = self.test_script_results
+                        .where(test_script_id: test_script_id)
+                        .first
```

Comment:
There are two interesting changes here.

1. `first` instead of `last`. Is my understanding of the default order wrong? The mark should be based on the most recent test script result. Maybe just an explicit order in the association is necessary.

2. No `nil` check. I don't know that we can assume at least one test script result exists when this method is called.